python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -612,6 +612,50 @@ def __disableEdits( self, pathListing ) :
 					if isinstance( source, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) :
 						source["enabled"].setValue( False )
 
+	def __selectedSetExpressions( self, pathListing ) :
+
+		result = set()
+
+		with Gaffer.Context( self.getContext() ) as context :
+			renderPassPath = pathListing.getPath().copy()
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if (
+					not columnSelection.isEmpty() and (
+						not isinstance( column, _GafferSceneUI._RenderPassEditor.OptionInspectorColumn ) or
+						not (
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetName" ) or
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetNames" ) or
+							Gaffer.Metadata.value( "option:" + column.inspector().name(), "ui:scene:acceptsSetExpression" )
+						)
+					)
+				) :
+					# We only return set expressions if all selected paths are in
+					# columns that accept set names or set expressions.
+					return []
+
+				for path in columnSelection.paths() :
+					renderPassPath.setFromString( path )
+					renderPassName = renderPassPath.property( "renderPassPath:name" )
+					if not renderPassName :
+						# We only return set expressions if all selected paths are render passes.
+						return []
+
+					context["renderPass"] = renderPassName
+					inspection = column.inspector().inspect()
+					if inspection.value() :
+						result.add( inspection.value().value )
+
+		return list( result )
+
+	def __selectAffected( self, pathListing ) :
+
+		result = IECore.PathMatcher()
```

Comment:
I think we need to scope our context here.

There's also a question of whether or not we should also set `${renderPass}` to correspond to the selected row(s), which would matter if scene content varied between render passes. Are we selecting from the current render pass for all selected passes, or are we selecting from the passes corresponding to the selection? I guess selecting in the current pass probably makes most sense, since that's likely the only one you can see in the Viewer and HierarchyView?

Reply:
Context scoped in b29f4e7 with `${renderPass}` set for the appropriate row when evaluating the set expressions for that row as users could conceivably modify set membership based on `${renderPass}`. In the extreme example case a `RENDER` set could be used for Camera Inclusions of all render passes, with its contents modified for each pass...